swupd/delta.go

```diff
@@ -0,0 +1,164 @@
+package swupd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/clearlinux/mixer-tools/helpers"
+	"github.com/pkg/errors"
+)
+
+// CreateDeltas creates all delta files between the previous and current
+// version of the supplied manifest, returning a list of files that
+// delta creation failed for.
+func CreateDeltas(manifest *Manifest, statedir string, from, to uint32) ([]*File, error) {
+	var c config
+	c, err := getConfig(statedir)
+	if err != nil {
+		return nil, err
+	}
+
+	var oldManifest *Manifest
+	var newManifest *Manifest
+
+	if oldManifest, err = ParseManifestFile(filepath.Join(c.outputDir, string(from), manifest.Name)); err != nil {
```

Comment:
Usually (if read from disk with `createManifestRecord()`) `manifest.Name` will not contain the `Manifest.` part of the name but will only be, for instance, `full` or `os-core` or the like. This isn't enforced so if `manifest.Name` is set to the full file name before calling this function it will be fine, but we should probably document via code comments in manifest.go what that field is expected to contain, then follow that for consistency.